File: Introductory Problems/Coin Piles/code.py
for _ in range(int(input())):
    a,b = map(int,input().split())
    if a > 2 * b or b > 2*a:
        print("NO")
        continue
    a = a%3
    b = b%3
    if (a==1 and b==2) or (a==2 and b==1) or (a==0 and b==0):
        print("YES")
    else:
        print("NO")

File: Introductory Problems/Coin Piles/code.js
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').trim().split('\n');
let n = parseInt(input[0], 10);

for(let i = 1; i<=n; i++ ){
    const line = input[i].split(' ').map(Number);
    let a = line[0], b = line[1];
    if (a>2*b || b>2*a) {
        console.log("NO"); continue;
    }
    a = a%3;
    b = b%3;
    if ((a==1 && b==2) || (a==2 && b==1) || (a==0 && b==0)) {
        console.log("YES");
    } else {
        console.log("NO");
    }
}